packages/lodestar/src/network/gossip/handlers/index.ts

```diff
@@ -93,29 +86,35 @@ export function getGossipHandlers(modules: ValidatorFnsModules): GossipHandlers
         throw e;
       }
 
-      // Handler
-      try {
-        await chain.processBlock(signedBlock);
-        // Returns the delay between the start of `block.slot` and `current time`
-        const delaySec = Date.now() / 1000 - (chain.genesisTime + slot * config.SECONDS_PER_SLOT);
-        metrics?.gossipBlock.elappsedTimeTillProcessed.observe(delaySec);
-      } catch (e) {
-        if (e instanceof BlockError) {
-          switch (e.type.code) {
-            case BlockErrorCode.ALREADY_KNOWN:
-            case BlockErrorCode.PARENT_UNKNOWN:
-            case BlockErrorCode.PRESTATE_MISSING:
-              break;
-            default:
-              network.peerRpcScores.applyAction(
-                PeerId.createFromB58String(peerIdStr),
-                PeerAction.LowToleranceError,
-                "BadGossipBlock"
-              );
+      // Handler - MUST NOT `await`, to allow validation result to be propagated
+
+      metrics?.registerBeaconBlock(OpSource.gossip, seenTimestampSec, signedBlock.message);
+
+      // `validProposerSignature = true`, in gossip validation the proposer signature is checked
+      chain
```

Comment:
Gossip handlers must not be awaited. This nuance could be enforced latter with a better architecture